java/ql/test/query-tests/Telemetry/AutomodelApplicationModeExtractCandidates/Test.java

```diff
@@ -0,0 +1,41 @@
+package com.github.codeql.test;
+
+import java.io.InputStream;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+class AutomodelApplicationModeExtractCandidates {
```

Comment:
High-level comment: Does it make sense to have one test directory, with a source file, and then 3 qlref files (pointing to each of the 3 queries)? Then for the same test source file we will see what gets considered a positive example, negative example, or candidate. This could save you some source code and library stub duplication across the test cases, while also allowing us to see in one place what the different candidates are.

Reply:
I think that's a great question/suggestion. When starting, my gut feeling was that it's a good idea to follow established conventions in the pre-existing tests (one suite per .ql file), and that we could get better readability from keeping things apart — but it turns out that the examples I used ended up being pretty similar. I've now pushed two commits to merge the three application-mode tests into one and the three framework-mode tests into one 👌